### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by davidanthoff**
Drop julia 0.6 support, remove use of Compat, adopt Project.toml.

I would also suggest that we tag the new release as v1.0.0: I assume this package is done, i.e. there won't be any new stuff added. So we might as well go to a proper semver version number.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name = "Nullables"
+uuid = "4d1e1d77-625e-5b40-9113-a560ec7a8ecd"
+version = "1.0.0-DEV"
```
Comment (nalimilan) [inline Project.toml]:
```suggestion
version = "1.0.0"
```

**Comment by davidanthoff** [comment]
Bump, I updated things a bit more and this should be ready to be merged. The Julia 1.3 32 bit build is failing, but that is due to a bug in the RC of Julia, not this package.

Also, happy to do the maintenance chores for this package if someone gives me write access.

**Comment by davidanthoff** [comment]
Bump, any chance we could move forward with this? CC @nalimilan

File: .travis.yml [hunk 2]
```diff
@@ -1,30 +1,51 @@
...
-after_success:
-  # push coverage results to Coveralls
-  - julia -e '
-    VERSION >= v"0.7.0-DEV.3656" && using Pkg
-    VERSION >= v"0.7.0-DEV.5183" || cd(Pkg.dir("Nullables"))
-    Pkg.add("Coverage")
-    using Coverage
-    Coveralls.submit(Coveralls.process_folder())
-    Codecov.submit(Codecov.process_folder())'
+jobs:
```
Comment (nalimilan) [inline .travis.yml:26]: Is this really needed? Other packages don't list this AFAIK.

File: Project.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+name = "Nullables"
+uuid = "4d1e1d77-625e-5b40-9113-a560ec7a8ecd"
+version = "1.0.0"
```
Comment (nalimilan) [inline Project.toml:3]: Bumping version from 0.0.8 to 1.0.0 means that some packages will have to change they version requirements. Is that really worth it? Do you plan to make further changes to this package (apart from maintenance fixes)?

File: .travis.yml (hunk 2 above)
Comment (davidanthoff) [reply .travis.yml:26]: I do want to make sure we test 32 bit Windows, and that seems to be the way to do that? I've seen a fair number of travis configs that test on Windows, but only 64 bit.

Having said that, I think we're pretty close to getting the GitHub Actions CI story finished, at which I would switch things over to that, so this is probably not a very permanent situation in any case.

File: Project.toml (hunk 3 above)
Comment (davidanthoff) [reply Project.toml:3]: No plans for any further changes, other than keeping it going.

As far as I understand it, though, any version lower than 0.1 is considered breaking by the Julia version semantics... So even if we increased this to `0.0.9`, I think all depending packages would have to change their bound. Or to say it another way, I think that for 0.0.x versions, even patch releases are considered breaking by Julia. So if we need to break things in any case, we might as well go to 1.0.0 and then leave this whole pre 1.0.0 sadness behind us, right?

I'll make sure to adjust bounds in the registry for depending packages and will also open PRs against the packages that depend on `Nullables` to adjust their version ranges once we have a new release tagged.

Comment (nalimilan) [reply Project.toml:3]: Ah, right, because of the `0.0`... so I guess 1.0 is OK.

File: .travis.yml (hunk 2 above)
Comment (nalimilan) [reply .travis.yml:26]: That sounds quite suboptimal, but well...

**Comment by nalimilan** [comment]
https://github.com/JuliaRegistries/General/pull/5627

**Comment by davidanthoff** [comment]
Great, thanks!

**Pull request merged**